[core]/es_extended/client/main.lua

```diff
@@ -57,66 +57,61 @@ AddEventHandler('esx:playerLoaded', function(xPlayer, isNew, skin)
 		NetworkSetFriendlyFireOption(true)
 	end
 
-	CreateThread(function()
-		local SetPlayerHealthRechargeMultiplier = SetPlayerHealthRechargeMultiplier
-		local BlockWeaponWheelThisFrame = BlockWeaponWheelThisFrame
-		local DisableControlAction = DisableControlAction
-		local IsPedArmed = IsPedArmed
-		local SetPlayerLockonRangeOverride = SetPlayerLockonRangeOverride
-		local DisablePlayerVehicleRewards = DisablePlayerVehicleRewards
-		local RemoveAllPickupsOfType = RemoveAllPickupsOfType
-		local HideHudComponentThisFrame = HideHudComponentThisFrame
-		local PlayerId = PlayerId()
-		local DisabledComps = {}
-		for i=1, #(Config.RemoveHudCommonents) do
-			if Config.RemoveHudCommonents[i] then
-				DisabledComps[#DisabledComps + 1] = i
-			end
-		end
+	local playerId = PlayerId()
 
-		while true do 
-			local Sleep = true
+	-- RemoveHudCommonents
+	for i=1, #(Config.RemoveHudCommonents) do
+		if Config.RemoveHudCommonents[i] then
+			SetHudComponentPosition(i, 999999.0, 999999.0)
+		end
+	end
 
-			if Config.DisableHealthRegeneration then
-				Sleep = false
-				SetPlayerHealthRechargeMultiplier(PlayerId, 0.0)
-			end
+	-- DisableNPCDrops
+	if Config.DisableNPCDrops then
+		local weaponPickups = {`PICKUP_WEAPON_CARBINERIFLE`, `PICKUP_WEAPON_PISTOL`, `PICKUP_WEAPON_PUMPSHOTGUN`}
+		for i = 1, #weaponPickups do 
+			ToggleUsePickupsForPlayer(playerId, weaponPickups[i], false) 
+		end
+	end
 
-			if Config.DisableWeaponWheel then
-				Sleep = false
-				BlockWeaponWheelThisFrame()
-				DisableControlAction(0, 37,true)
-			end
+	-- DisableVehicleRewards
+	if Config.DisableVehicleRewards then
+		AddEventHandler('esx:enteredVehicle', function(vehicle, plate, seat, displayName, netId)
```

Comment:
I think it's enough to look at the emergency vehicle class or other vehicle class can you give a reward?